src/cascadia/TerminalSettingsEditor/Profiles.xaml

```diff
@@ -769,9 +769,20 @@
                                                 ClearSettingValue="{x:Bind State.Profile.ClearBellStyle}"
                                                 HasSettingValue="{x:Bind State.Profile.HasBellStyle, Mode=OneWay}"
                                                 SettingOverrideSource="{x:Bind State.Profile.BellStyleOverrideSource, Mode=OneWay}">
-                            <muxc:RadioButtons ItemTemplate="{StaticResource EnumRadioButtonTemplate}"
-                                               ItemsSource="{x:Bind BellStyleList, Mode=OneWay}"
-                                               SelectedItem="{x:Bind CurrentBellStyle, Mode=TwoWay}" />
+                            <StackPanel>
+                                <CheckBox x:Name="AudibleCheckBox"
+                                          x:Uid="Profile_BellStyleAudible"
+                                          Click="BellStyle_SelectionChanged"
+                                          IsChecked="{x:Bind BellStyleAudible, Mode=OneWay}" />

```

Comment:
Here's a very clever trick you can do.

You can replace the OneWay binding *and* the click handler with a two-way function binding.

Add the following functions to your IDL:

```
void IsBellStyleFlagSet(UInt32 style);
void SetBellStyleAudible(Boolean on);
void SetBellStyleVisual(Boolean on);
void SetBellStyleWindow(Boolean on);
```

and then you can use a function bind like this:

```
<CheckBox Blah="Blah"
          IsChecked="{x:Bind IsBellStyleFlagSet(1), BindBack=SetBellStyleAudible, Mode=TwoWay}" />
```

It will set the checkbox value to the return value of `IsBellStyleFlagSet(1)` and then call `SetBellStyleAudible(true)` (or `...(false)`).

You can then send out a Property Changed Notification for `IsBellStyleFlagSet`.

Rinse/repeat for flag `2`, `4`, `8` and the other `SetBellStyle` functions.

Reply:
This would be a good underpinning for an eventual "repeated flag checkbox" control (but it would probably have to use a click handler again ;P)

Reply:
TIL `BindBack` is a thing, thank you!